### PR TITLE
Connectors - updates for supporting Office365 

### PR DESCRIPTION
**Pull request by LucGenetier**

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:x: 2 Binary Compatibility issues for **Microsoft.PowerFx.Core** compared against [version 0.2.2-preview.20220811-316446](https://www.nuget.org/packages/Microsoft.PowerFx.Core/0.2.2-preview.20220811-316446).
<details>

```
InterfacesShouldHaveSameMembers : Interface member 'Microsoft.PowerFx.Intellisense.IPowerFxScope.Check(System.String, Microsoft.PowerFx.ParserOptions)' is present in the implementation but not in the contract.
InterfacesShouldHaveSameMembers : Interface member 'Microsoft.PowerFx.Intellisense.IPowerFxScope.Suggest(System.String, System.Int32, Microsoft.PowerFx.ParserOptions)' is present in the implementation but not in the contract.
```

</details>
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by MikeStall — APPROVED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs [hunk 1]
```diff
@@ -138,7 +138,7 @@ public static bool HasDefaultValue(this OpenApiParameter param)
 
         // Internal parameters are not showen to the user. 
         // They can have a default value or be special cased by the infrastructure (like "connectionId").
-        public static bool IsInternal(this OpenApiParameter param) => param.Extensions.TryGetValue("x-ms-visibility", out _);
+        public static bool IsInternal(this OpenApiParameter param) => param.Extensions.TryGetValue("x-ms-visibility", out var openApiExt) && openApiExt is OpenApiString openApiStr && openApiStr.Value == "internal";
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs:141]: >openApiStr.Value == "internal [](http://example.com/codeflow?start=183&length=29)

This is the key part of the change.

**Pull request merged**